### DISCUSSION TITLE
What is behind the scene TicketBitMap ?

### DISCUSSION THREAD

**Discussion by perfogic** [Jeiwan/uniswapv3-book#68] [Milestone 2. Second Swap]
Hi, I just finished milestone 2 in my journey on learning uniswap v3. I see that when we swap, we can calculate nextPrice => deduce nextTick in python formula. 
<img width="588" alt="image" src="https://github.com/Jeiwan/uniswapv3-book/assets/74671798/fad822b2-7861-4868-b9ba-2b59bcf5ab8a">

But why next tick is calculated like this in TickBitMap. I doesn't see any relations between these formula: 
<img width="585" alt="image" src="https://github.com/Jeiwan/uniswapv3-book/assets/74671798/28320a99-feb1-4fee-ae89-84b1eb468d5d">

**Comment by perfogic** [comment]
Solved

**Comment by lylaly098** [comment]
its perfect

**Comment by 0xJungFu** [comment]
Learing this too. Thx for experience
